pkg/services/rpcsrv/server_test.go

```diff
@@ -3331,3 +3332,21 @@ func BenchmarkHandleIn(b *testing.B) {
 {"type": "Integer", "value": "42"}, {"type": "Boolean", "value": false}]]}`))
 	})
 }
+
+func TestFailedPreconditionShutdown(t *testing.T) {
+	_, srv, _ := initClearServerWithCustomConfig(t, func(c *config.Config) {
+		c.ApplicationConfiguration.RPC.Addresses = []string{"not an address"}
+	})
+
+	srv.Start()
+	require.Positive(t, len(srv.errChan)) // this is how Start reports internal failures
+
+	var stopped atomic.Bool
+
+	go func() {
+		srv.Shutdown()
+		stopped.Store(true)
+	}()
+
+	require.Eventually(t, stopped.Load, 5*time.Second, 100*time.Millisecond, "Shutdown should return")
```

Comment:
5 seconds is really a lot, we have about 1 second at max for such tests.

Reply:
We better have safer value here, if it finishes in 100ms we're not missing anythin, if it takes 1500ms there is difference between failed and successful test. I prefer successful ones.